src/Bicep.Core/Configuration/ExperimentalFeaturesEnabled.cs

```diff
@@ -28,4 +30,51 @@ public static ExperimentalFeaturesEnabled Bind(JsonElement element)
         => element.ToNonNullObject<ExperimentalFeaturesEnabled>();
 
     public void WriteTo(Utf8JsonWriter writer) => JsonElementFactory.CreateElement(this).WriteTo(writer);
+
+    public static readonly ExperimentalFeaturesEnabled AllDisabled = new(
+        SymbolicNameCodegen: false,
+        ExtendableParamFiles: false,
+        ResourceTypedParamsAndOutputs: false,
+        SourceMapping: false,
+        LegacyFormatter: false,
+        TestFramework: false,
+        Assertions: false,
+        WaitAndRetry: false,
+        LocalDeploy: false,
+        ResourceInfoCodegen: false,
+        ModuleExtensionConfigs: false,
+        DesiredStateConfiguration: false,
+        ExternalInputFunction: false,
+        OnlyIfNotExists: false,
+        ModuleIdentity: false);
+
+    public IFeatureProvider ToFeatureProvider() => new FeatureProviderAdapter(this);
```

Comment:
Since this is only used by `EmptySemanticModel` to create a dummy feature provider, it would be better to define a private implementation of `IFeatureProvider` within `EmptySemanticModel` and remove `ToFeatureProvider`. Leaving `ToFeatureProvider` public risks accidental use elsewhere, where accessing `AssemblyVersion` or `CacheRootDirectory` could throw.